### PR TITLE
IDFF with PSSOFB

### PR DESCRIPTION
**Pull request by xresende**

### CODE REVIEW

File: siriuspy/siriuspy/idff/main.py [hunk 1]
```diff
@@ -134,6 +147,23 @@ def set_config_name(self, value):
...
+        if self._loop_state == self._const.LoopState.Closed:
+            self._update_log('ERR:Open loop before changing configuration.')
+            return False
+
+        if not self._idff_prepare_corrs_state(value):
+            self._update_log(('ERR:Could not configure IDFF correctors '
+                'when changing SOFBMode.'))
+            return False
+
+        self._pssofb_isused = bool(value)
```
Comment (anacso17) [inline siriuspy/siriuspy/idff/main.py:164]: Missing update SOFBMode-Sts

File: siriuspy/siriuspy/idff/main.py [hunk 2]
```diff
@@ -250,3 +305,64 @@ def _do_ff(self):
...
+
+        # build bsmpid -> psname dict
+        bsmp_devs = dict()
+        for devices in pssofb.bbb2devs.values():
+            for devname, bsmpid in devices:
+                bsmp_devs[bsmpid] = devname
+
+        return pssofb, bsmp_devs
+
+    def _pssfob_current_setpoint(self, setpoints, corrdevs):
```
Comment (anacso17) [inline siriuspy/siriuspy/idff/main.py:361]: maybe rename to `_pssfob_get_current_setpoint`

File: siriuspy/siriuspy/idff/main.py (hunk 1 above)
Comment (xresende) [reply siriuspy/siriuspy/idff/main.py:164]: done.

File: siriuspy/siriuspy/idff/main.py [hunk 3]
```diff
@@ -250,3 +305,64 @@ def _do_ff(self):
...
+
+        # turn on
+        for dev in corrdevs:
+            if not dev.cmd_turn_on():
+                return False
+
+        # opmode slowref
+        for dev in corrdevs:
+            if not dev.cmd_slowref():
+                return False
```
[the remark this answers is not in the document]
Comment (xresende) [reply siriuspy/siriuspy/idff/main.py:326]: maybe a -Cmd PV in the future that configures the IDFF correctors, like SOFB does?

[the remark this answers is not in the document]
Comment (xresende) [reply siriuspy/siriuspy/idff/main.py:326]: changed

File: siriuspy/siriuspy/idff/main.py (hunk 2 above)
Comment (xresende) [reply siriuspy/siriuspy/idff/main.py:361]: done.

File: siriuspy/siriuspy/idff/main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (anacso17) [reply siriuspy/siriuspy/idff/main.py]: Yes, maybe we could implement an auxiliary command that ensure the desired power supply state, but I think we could avoid sending several commands when they are not necessary, like in this case

**Review by fernandohds564 — APPROVED** [review]

**Pull request merged**